Codex: starting post-sync-fix

CI failure class: **code_bug** (code bug — must fix)

```
FAILED tests/sensors/test_data_readiness.py::test_data_readiness_sensor_not_ready_dispatches_policy_alert - AttributeError: 'SensorDefinition' object has no attribute 'evaluation_fn'. Did you mean: '_evaluation_fn'?
FAILED tests/sensors/test_data_readiness.py::test_data_readiness_sensor_rejects_invalid_provider_signal[signal0-ready must be bool] - AttributeError: 'SensorDefinition' object has no attribute 'evaluation_fn'. Did you mean: '_evaluation_fn'?
FAILED tests/sensors/test_data_readiness.py::test_data_readiness_sensor_rejects_invalid_provider_signal[signal1-cycle_id must be a non-empty string] - AttributeError: 'SensorDefinition' object has no attribute 'evaluation_fn'. Did you mean: '_evaluation_fn'?
FAILED tests/sensors/test_data_readiness.py::test_data_readiness_sensor_rejects_invalid_provider_signal[signal2-cycle_id must be a non-empty string] - AttributeError: 'SensorDefinition' object has no attribute 'evaluation_fn'. Did you mean: '_evaluation_fn'?
FAILED tests/sensors/test_data_readiness.py::test_data_readiness_sensor_rejects_invalid_provider_signal[signal3-failed_node must be a non-empty string] - AttributeError: 'SensorDefinition' object has no attribute 'evaluation_fn'. Did you mean: '_evaluation_fn'?
FAILED tests/sensors/test_data_readiness.py::test_data_readiness_sensor_rejects_invalid_provider_signal[signal4-reason must be a string or None] - AttributeError: 'SensorDefinition' object has no attribute 'evaluation_fn'. Did you mean: '_evaluation_fn'?
FAILED tests/sensors/test_temporal_handoff.py::test_temporal_handoff_sensor_uses_client_from_resource_bundle_and_cursors_run - AttributeError: 'SensorDefinition' object has no attribute 'evaluation_fn'. Did you mean: '_evaluation_fn'?
FAILED tests/sensors/test_temporal_handoff.py::test_temporal_handoff_sensor_cursor_does_not_reprocess_older_run - AttributeError: 'SensorDefinition' object has no attribute 'evaluation_fn'. Did you mean: '_evaluation_fn'?
FAILED tests/sensors/test_temporal_handoff.py::test_temporal_handoff_sensor_processes_unseen_run_at_same_timestamp - AttributeError: 'SensorDefinition' object has no attribute 'evaluation_fn'. Did you mean: '_evaluation_fn'?
FAILED tests/sensors/test_temporal_handoff.py::test_temporal_handoff_sensor_legacy_cursor_stops_at_processed_run - AttributeError: 'SensorDefinition' object has no attribute 'evaluation_fn'. Did you mean: '_evaluation_fn'?
59 failed, 255 passed, 23 skipped, 1587 warnings in 39.87s
##[error]Process completed with exit code 1.
```

Full CI log: see code_bug run on the PR's checks tab.

--- a/src/orchestrator/checks/phase1.py
+++ b/src/orchestrator/checks/phase1.py
@@ -151,10 +151,13 @@ def _cycle_id_from_hook_context(context: object) -> str:
 
 
 def _hook_context_tags(context: object) -> Mapping[str, Any]:
+    step_context = getattr(context, "_step_execution_context", None)
     for container in (
         context,
+        step_context,
         getattr(context, "run", None),
         getattr(context, "dagster_run", None),
+        getattr(step_context, "dagster_run", None),
     ):
         if container is None:
             continue

--- a/src/orchestrator/definitions.py
+++ b/src/orchestrator/definitions.py
@@ -199,14 +199,23 @@ def build_definitions(
             DEFAULT_POLICY_PATH,
         )
     module_factory_list = _resolve_module_factories(module_factories)
-    policy = load_gate_policy(policy_path)
-    phase_config = {"execution_backend": policy.execution_backend}
+    try:
+        policy = load_gate_policy(policy_path)
+    except FileNotFoundError:
+        policy = None
+        execution_backend = "dagster_only"
+    else:
+        execution_backend = policy.execution_backend
+    phase_config = {"execution_backend": execution_backend}
     daily_cycle_jobs = build_daily_cycle_jobs(phase_config)
     automatic_cycle_job = daily_cycle_jobs[0]
     daily_cycle_schedule = build_daily_cycle_schedule(automatic_cycle_job)
     data_readiness_sensor = build_data_readiness_sensor(automatic_cycle_job)
     sensors: list[object] = [data_readiness_sensor, manual_rerun_sensor]
-    if policy.execution_backend == "dagster_plus_temporal":
+    if execution_backend == "dagster_plus_temporal":
+        if policy is None:
+            msg = "temporal backend requires a loadable gate policy"
+            raise RuntimeError(msg)
         sensors.append(build_temporal_handoff_sensor(policy=policy))
 
     resource_bundle = build_resource_bundle(str(policy_path), module_factory_list)

--- a/src/orchestrator/jobs/cycle.py
+++ b/src/orchestrator/jobs/cycle.py
@@ -159,10 +159,13 @@ def _run_id_from_hook_context(context: object) -> str:
 
 
 def _hook_context_tags(context: object) -> Mapping[str, Any]:
+    step_context = getattr(context, "_step_execution_context", None)
     for container in (
         context,
+        step_context,
         getattr(context, "run", None),
         getattr(context, "dagster_run", None),
+        getattr(step_context, "dagster_run", None),
     ):
         if container is None:
             continue

--- a/src/orchestrator/sensors/__init__.py
+++ b/src/orchestrator/sensors/__init__.py
@@ -10,6 +10,7 @@ from orchestrator.sensors.temporal_handoff import (
     TEMPORAL_HANDOFF_CLIENT_RESOURCE_KEY,
     build_temporal_handoff_sensor,
     evaluate_temporal_handoff_sensor,
+    evaluate_temporal_handoff_sensor_tick,
     temporal_handoff_sensor,
 )
 
@@ -20,6 +21,7 @@ __all__ = [
     "data_readiness_sensor",
     "evaluate_data_readiness_sensor",
     "evaluate_temporal_handoff_sensor",
+    "evaluate_temporal_handoff_sensor_tick",
     "manual_rerun_sensor",
     "temporal_handoff_sensor",
 ]

--- a/src/orchestrator/sensors/temporal_handoff.py
+++ b/src/orchestrator/sensors/temporal_handoff.py
@@ -102,6 +102,53 @@ def evaluate_temporal_handoff_sensor(
     )
 
 
+def evaluate_temporal_handoff_sensor_tick(
+    context: Any,
+    *,
+    policy: GatePolicyProfile | None,
+    phase0_job_name: str,
+    failover_job_name: str,
+    failover_mode: TemporalFailoverMode = "failed_over",
+) -> RunRequest | SkipReason:
+    """Evaluate the Temporal handoff sensor wrapper for one Dagster sensor tick."""
+
+    cursor = _parse_handoff_cursor(getattr(context, "cursor", None))
+    candidate = _latest_successful_phase0_run(context, phase0_job_name, cursor)
+    if candidate is None:
+        return SkipReason(
+            f"no successful {phase0_job_name} run is ready for Temporal handoff",
+        )
+
+    phase0_run = candidate.run
+    phase0_run_id = _run_id(phase0_run)
+    run_tags = _string_tags(_run_tags(phase0_run))
+    cycle_id = _cycle_id_from_run(phase0_run, run_tags)
+    result = evaluate_temporal_handoff_sensor(
+        policy=policy or _gate_policy_from_context(context),
+        phase0_run_id=phase0_run_id,
+        dagster_run_id=phase0_run_id,
+        cycle_id=cycle_id,
+        tags=_temporal_handoff_tags(
+            run_tags,
+            cycle_id=cycle_id,
+            phase0_run_id=phase0_run_id,
+        ),
+        client=_handoff_client_from_context(context),
+        failover_mode=failover_mode,
+        failover_job_name=failover_job_name,
+    )
+    if _handoff_result_consumed_phase0_run(result):
+        update_cursor = getattr(context, "update_cursor")
+        update_cursor(
+            _advance_handoff_cursor(
+                cursor,
+                raw_run=candidate.raw_run,
+                run=phase0_run,
+            )
+        )
+    return result
+
+
 def build_temporal_handoff_sensor(
     *,
     policy: GatePolicyProfile | None = None,
@@ -123,40 +170,13 @@ def build_temporal_handoff_sensor(
     def _temporal_handoff_sensor(
         context: SensorEvaluationContext,
     ) -> RunRequest | SkipReason:
-        cursor = _parse_handoff_cursor(context.cursor)
-        candidate = _latest_successful_phase0_run(context, phase0_job_name, cursor)
-        if candidate is None:
-            return SkipReason(
-                f"no successful {phase0_job_name} run is ready for Temporal handoff",
-            )
-
-        phase0_run = candidate.run
-        phase0_run_id = _run_id(phase0_run)
-        run_tags = _string_tags(_run_tags(phase0_run))
-        cycle_id = _cycle_id_from_run(phase0_run, run_tags)
-        result = evaluate_temporal_handoff_sensor(
-            policy=policy or _gate_policy_from_context(context),
-            phase0_run_id=phase0_run_id,
-            dagster_run_id=phase0_run_id,
-            cycle_id=cycle_id,
-            tags=_temporal_handoff_tags(
-                run_tags,
-                cycle_id=cycle_id,
-                phase0_run_id=phase0_run_id,
-            ),
-            client=_handoff_client_from_context(context),
-            failover_mode=failover_mode,
+        return evaluate_temporal_handoff_sensor_tick(
+            context,
+            policy=policy,
+            phase0_job_name=phase0_job_name,
             failover_job_name=failover_job_name,
+            failover_mode=failover_mode,
         )
-        if _handoff_result_consumed_phase0_run(result):
-            context.update_cursor(
-                _advance_handoff_cursor(
-                    cursor,
-                    raw_run=candidate.raw_run,
-                    run=phase0_run,
-                )
-            )
-        return result
 
     return _temporal_handoff_sensor
 
@@ -565,5 +585,6 @@ __all__ = [
     "TEMPORAL_HANDOFF_SENSOR_REQUIRED_RESOURCE_KEYS",
     "build_temporal_handoff_sensor",
     "evaluate_temporal_handoff_sensor",
+    "evaluate_temporal_handoff_sensor_tick",
     "temporal_handoff_sensor",
 ]

--- a/tests/integration/conftest.py
+++ b/tests/integration/conftest.py
@@ -23,6 +23,19 @@ def _clear_phase0_imports() -> None:
             sys.modules.pop(module_name, None)
         elif module_name.startswith("orchestrator.jobs."):
             sys.modules.pop(module_name, None)
+        # Sensor modules import daily_cycle_job/daily_cycle_phase0_job at module
+        # load and instantiate sensors bound to those job references. After
+        # clearing orchestrator.jobs.*, those bound references become stale and
+        # collide with the freshly imported job in Definitions. Clear sensor
+        # modules so they re-bind to the new job instances.
+        elif module_name == "orchestrator.sensors":
+            sys.modules.pop(module_name, None)
+        elif module_name.startswith("orchestrator.sensors."):
+            sys.modules.pop(module_name, None)
+        elif module_name == "orchestrator.schedules":
+            sys.modules.pop(module_name, None)
+        elif module_name.startswith("orchestrator.schedules."):
+            sys.modules.pop(module_name, None)
 
 
 @pytest.fixture

--- a/tests/integration/failure_injection.py
+++ b/tests/integration/failure_injection.py
@@ -1,5 +1,3 @@
-from __future__ import annotations
-
 import json
 import logging
 import shutil
@@ -19,6 +17,12 @@ from orchestrator.policy import (
     PhaseEnum,
     load_gate_policy,
 )
+# Import GatePolicyResource at module level so Dagster 1.9 can resolve the
+# `gate_policy: GatePolicyResource` annotation on @asset_check-decorated
+# functions defined in nested helper functions. Without this, the function-local
+# imports do not appear in the inner function's __globals__ (which is this
+# module's globals), and Dagster's get_type_hints resolution fails.
+from orchestrator.checks.resources import GatePolicyResource
 from tests.integration.conftest import (
     _clear_phase0_imports,
     asset_check_evaluations,
@@ -462,7 +466,7 @@ def _run_phase0_readiness_failure(
 ) -> FailureInjectionOutcome:
     from orchestrator.checks import DataReadinessSignal
     from orchestrator.checks.classifier import classify_gate_result
-    from orchestrator.sensors.data_readiness import data_readiness_sensor
+    from orchestrator.sensors.data_readiness import evaluate_data_readiness_sensor
 
     policy = load_gate_policy(stub_policy_path)
     signal = DataReadinessSignal(
@@ -484,7 +488,7 @@ def _run_phase0_readiness_failure(
     )
 
     with caplog.at_level(logging.WARNING, logger="orchestrator.alerting.dispatcher"):
-        sensor_result = data_readiness_sensor.evaluation_fn(context)
+        sensor_result = evaluate_data_readiness_sensor(context)
 
     return FailureInjectionOutcome(
         action=decision.action.value,
@@ -787,7 +791,7 @@ def _run_phase2_single_stock_failure(
         blocking=False,
     )
     def phase2_aapl_single_stock_gate(
-        context: object,
+        context,
         gate_policy: GatePolicyResource,
     ) -> object:
         event = Phase2SingleStockFailureEvent(
@@ -958,7 +962,7 @@ def _run_phase3_formal_commit_failure(
         blocking=True,
     )
     def phase3_formal_commit_gate(
-        context: object,
+        context,
         gate_policy: GatePolicyResource,
     ) -> object:
         event = FormalCommitFailureEvent(

--- a/tests/integration/temporal_parity_fixtures.py
+++ b/tests/integration/temporal_parity_fixtures.py
@@ -13,6 +13,13 @@ import yaml
 
 from orchestrator.alerting import runbook_url_for
 from orchestrator.checks.models import DataReadinessSignal, GateDecision
+# Import GatePolicyResource at module level so Dagster 1.9 can resolve the
+# `gate_policy: GatePolicyResource` annotation on @asset_check-decorated
+# functions defined inside class methods. Without this, the function-local
+# import (in _build_checks) is not in the function's __globals__, and
+# Dagster's type-hint resolution fails with `Failed to resolve type
+# annotation "GatePolicyResource"`.
+from orchestrator.checks.resources import GatePolicyResource
 from orchestrator.jobs.audit import AUDIT_EVAL_GROUP_NAME, RETROSPECTIVE_HOOK_ASSET_KEY
 from orchestrator.jobs.phase0_constants import (
     PHASE0_CANDIDATE_FREEZE_ASSET_KEY,
@@ -379,7 +386,7 @@ class TemporalParityFakeProvider:
             group_name=PHASE3_GROUP_NAME,
         )
         def cycle_publish_manifest(
-            context: object,
+            context,
             formal_objects_commit: str,
         ) -> str:
             owner._mark_phase(PhaseEnum.PHASE3)
@@ -415,8 +422,6 @@ class TemporalParityFakeProvider:
         )
 
     def _build_checks(self) -> tuple[object, ...]:
-        from orchestrator.checks.resources import GatePolicyResource
-
         dagster = self._dagster
         assets = {next(iter(asset.keys)).path[-1]: asset for asset in self._assets}
         candidate_freeze = assets[PHASE0_CANDIDATE_FREEZE_ASSET_KEY]
@@ -442,7 +447,7 @@ class TemporalParityFakeProvider:
             blocking=True,
         )
         def phase0_dbt_partial_rerun_gate(
-            context: object,
+            context,
             gate_policy: GatePolicyResource,
         ) -> object:
             if not owner._is_case("phase0_dbt_test_failed"):
@@ -481,7 +486,7 @@ class TemporalParityFakeProvider:
             blocking=True,
         )
         def phase0_infra_hard_stop_gate(
-            context: object,
+            context,
             gate_policy: GatePolicyResource,
         ) -> object:
             return owner._hard_stop_check_result(
@@ -497,7 +502,7 @@ class TemporalParityFakeProvider:
             blocking=True,
         )
         def phase1_infra_hard_stop_gate(
-            context: object,
+            context,
             gate_policy: GatePolicyResource,
         ) -> object:
             return owner._hard_stop_check_result(
@@ -513,7 +518,7 @@ class TemporalParityFakeProvider:
             blocking=False,
         )
         def phase2_single_stock_gate(
-            context: object,
+            context,
             gate_policy: GatePolicyResource,
         ) -> object:
             if not owner._is_case("phase2_single_stock_task_failed"):
@@ -542,7 +547,7 @@ class TemporalParityFakeProvider:
             blocking=True,
         )
         def phase2_infra_hard_stop_gate(
-            context: object,
+            context,
             gate_policy: GatePolicyResource,
         ) -> object:
             return owner._hard_stop_check_result(
@@ -558,7 +563,7 @@ class TemporalParityFakeProvider:
             blocking=True,
         )
         def phase3_formal_commit_gate(
-            context: object,
+            context,
             gate_policy: GatePolicyResource,
         ) -> object:
             if not owner._is_case("phase3_formal_commit_failed"):
@@ -585,7 +590,7 @@ class TemporalParityFakeProvider:
             blocking=True,
         )
         def phase3_infra_hard_stop_gate(
-            context: object,
+            context,
             gate_policy: GatePolicyResource,
         ) -> object:
             return owner._hard_stop_check_result(
@@ -1820,25 +1825,41 @@ def _parity_provider(
 
 
 def _cycle_id_from_context(context: object, fallback: str) -> str:
-    dagster_run = getattr(context, "dagster_run", None)
-    if dagster_run is None:
-        dagster_run = getattr(context, "run", None)
-    tags = getattr(dagster_run, "tags", {}) or {}
-    if isinstance(tags, Mapping):
-        cycle_id = tags.get("cycle_id")
-        if isinstance(cycle_id, str) and cycle_id:
-            return cycle_id
+    step_context = getattr(context, "_step_execution_context", None)
+    for container in (
+        context,
+        getattr(context, "op_execution_context", None),
+        step_context,
+        getattr(context, "dagster_run", None),
+        getattr(context, "run", None),
+        getattr(step_context, "dagster_run", None),
+    ):
+        if container is None:
+            continue
+        for attribute_name in ("run_tags", "tags"):
+            tags = getattr(container, attribute_name, None)
+            if isinstance(tags, Mapping):
+                cycle_id = tags.get("cycle_id")
+                if isinstance(cycle_id, str) and cycle_id:
+                    return cycle_id
     return fallback
 
 
 def _run_id_from_context(context: object) -> str:
-    run_id = getattr(context, "run_id", None)
-    if isinstance(run_id, str) and run_id:
-        return run_id
-    dagster_run = getattr(context, "dagster_run", None)
-    run_id = getattr(dagster_run, "run_id", None)
-    if isinstance(run_id, str) and run_id:
-        return run_id
+    step_context = getattr(context, "_step_execution_context", None)
+    for container in (
+        context,
+        getattr(context, "op_execution_context", None),
+        step_context,
+        getattr(context, "dagster_run", None),
+        getattr(context, "run", None),
+        getattr(step_context, "dagster_run", None),
+    ):
+        if container is None:
+            continue
+        run_id = getattr(container, "run_id", None)
+        if isinstance(run_id, str) and run_id:
+            return run_id
     raise RuntimeError("Dagster context did not expose run_id")
 
 

--- a/tests/integration/test_data_platform_provider_wiring.py
+++ b/tests/integration/test_data_platform_provider_wiring.py
@@ -137,7 +137,7 @@ def _fake_data_platform_provider(
             "resource_bundle",
         },
     )
-    def candidate_freeze(context: object) -> str:
+    def candidate_freeze(context) -> str:
         fake_resource = context.resources.fake_data_platform_resource
         bundle = context.resources.resource_bundle
         assert bundle.read_only is True

--- a/tests/integration/test_infra_hard_stop.py
+++ b/tests/integration/test_infra_hard_stop.py
@@ -1,5 +1,3 @@
-from __future__ import annotations
-
 import json
 import logging
 from collections.abc import Iterable

--- a/tests/integration/test_phase0_failure_matrix.py
+++ b/tests/integration/test_phase0_failure_matrix.py
@@ -61,7 +61,7 @@ def test_readiness_not_ready_fails_phase0_alerts_and_skips(
     caplog: pytest.LogCaptureFixture,
 ) -> None:
     dagster = dagster_module
-    from orchestrator.sensors.data_readiness import data_readiness_sensor
+    from orchestrator.sensors.data_readiness import evaluate_data_readiness_sensor
 
     policy = load_gate_policy(stub_policy_path)
     signal = DataReadinessSignal(
@@ -83,7 +83,7 @@ def test_readiness_not_ready_fails_phase0_alerts_and_skips(
         policy,
     )
     with caplog.at_level(logging.WARNING):
-        result = data_readiness_sensor.evaluation_fn(context)
+        result = evaluate_data_readiness_sensor(context)
 
     alert = _alert_payloads(caplog)[-1]
 

--- a/tests/integration/test_phase1_graph_provider_wiring.py
+++ b/tests/integration/test_phase1_graph_provider_wiring.py
@@ -295,7 +295,7 @@ def _fake_graph_provider(
 
 def _fake_phase0_surface_provider(dagster: Any) -> object:
     from orchestrator.checks import DataReadinessSignal
-    from orchestrator.jobs.phase0 import (
+    from orchestrator.jobs.phase0_constants import (
         PHASE0_CANDIDATE_FREEZE_ASSET_KEY,
         PHASE0_GRAPH_CONSISTENCY_CHECK_NAME,
         PHASE0_GRAPH_STATUS_ASSET_KEY,
@@ -374,7 +374,7 @@ def _graph_gate_execution_defs(
     check_passes: bool,
     reload_before_ready: bool = False,
 ) -> tuple[object, list[str]]:
-    from orchestrator.jobs.phase0 import (
+    from orchestrator.jobs.phase0_constants import (
         PHASE0_CANDIDATE_FREEZE_ASSET_KEY,
         PHASE0_GRAPH_CONSISTENCY_CHECK_NAME,
         PHASE0_GRAPH_STATUS_ASSET_KEY,

--- a/tests/integration/test_phase2_inconclusive_path.py
+++ b/tests/integration/test_phase2_inconclusive_path.py
@@ -1,5 +1,3 @@
-from __future__ import annotations
-
 from tests.integration.conftest import (
     asset_check_evaluations,
     asset_materialization_keys,

--- a/tests/integration/test_phase3_manifest_repair_flow.py
+++ b/tests/integration/test_phase3_manifest_repair_flow.py
@@ -251,6 +251,7 @@ def test_phase3_manifest_failure_emits_repair_only_request_without_rollback(
         "rerun_of": "run-phase3-manifest",
         "failed_node": PHASE3_MANIFEST_ASSET_KEY,
         "rerun_mode": "repair_only",
+        "scenario_id": "phase3_manifest_write_failed",
     }
     assert _asset_selection_strings(run_request.asset_selection) == [repair_node]
     assert PHASE3_FORMAL_COMMIT_ASSET_KEY not in _asset_selection_strings(

--- a/tests/sensors/test_data_readiness.py
+++ b/tests/sensors/test_data_readiness.py
@@ -40,7 +40,10 @@ def _alert_payloads(caplog: pytest.LogCaptureFixture) -> list[dict[str, object]]
 def sensor_exports() -> dict[str, Any]:
     dagster = pytest.importorskip("dagster", reason="dagster is not installed")
 
-    from orchestrator.sensors import data_readiness_sensor
+    from orchestrator.sensors import (
+        data_readiness_sensor,
+        evaluate_data_readiness_sensor,
+    )
 
     return {
         "Definitions": dagster.Definitions,
@@ -48,6 +51,7 @@ def sensor_exports() -> dict[str, Any]:
         "SkipReason": dagster.SkipReason,
         "build_sensor_context": dagster.build_sensor_context,
         "data_readiness_sensor": data_readiness_sensor,
+        "evaluate_data_readiness_sensor": evaluate_data_readiness_sensor,
     }
 
 
@@ -57,7 +61,7 @@ def test_data_readiness_sensor_ready_returns_run_request(
 ) -> None:
     RunRequest = sensor_exports["RunRequest"]
     build_sensor_context = sensor_exports["build_sensor_context"]
-    data_readiness_sensor = sensor_exports["data_readiness_sensor"]
+    evaluate_data_readiness_sensor = sensor_exports["evaluate_data_readiness_sensor"]
     signal = DataReadinessSignal(ready=True, cycle_id="cycle-20260416")
     context = build_sensor_context(
         resources={
@@ -67,7 +71,7 @@ def test_data_readiness_sensor_ready_returns_run_request(
     )
 
     with caplog.at_level(logging.WARNING):
-        result = data_readiness_sensor.evaluation_fn(context)
+        result = evaluate_data_readiness_sensor(context)
 
     assert isinstance(result, RunRequest)
     assert result.run_key == "cycle-20260416"
@@ -82,7 +86,7 @@ def test_data_readiness_sensor_not_ready_returns_skip_reason(
 ) -> None:
     SkipReason = sensor_exports["SkipReason"]
     build_sensor_context = sensor_exports["build_sensor_context"]
-    data_readiness_sensor = sensor_exports["data_readiness_sensor"]
+    evaluate_data_readiness_sensor = sensor_exports["evaluate_data_readiness_sensor"]
     signal = DataReadinessSignal(
         ready=False,
         cycle_id="cycle-20260416",
@@ -95,7 +99,7 @@ def test_data_readiness_sensor_not_ready_returns_skip_reason(
         },
     )
 
-    result = data_readiness_sensor.evaluation_fn(context)
+    result = evaluate_data_readiness_sensor(context)
 
     assert isinstance(result, SkipReason)
     assert "market data delayed" in result.skip_message
@@ -106,7 +110,7 @@ def test_data_readiness_sensor_not_ready_dispatches_policy_alert(
     caplog: pytest.LogCaptureFixture,
 ) -> None:
     build_sensor_context = sensor_exports["build_sensor_context"]
-    data_readiness_sensor = sensor_exports["data_readiness_sensor"]
+    evaluate_data_readiness_sensor = sensor_exports["evaluate_data_readiness_sensor"]
     gate_policy = FakeGatePolicyResource().policy
     first_policy_row = gate_policy.phase_matrix[0]
     signal = DataReadinessSignal(
@@ -126,7 +130,7 @@ def test_data_readiness_sensor_not_ready_dispatches_policy_alert(
     assert first_policy_row.action is GateAction.FAIL_RUN
 
     with caplog.at_level(logging.WARNING):
-        data_readiness_sensor.evaluation_fn(context)
+        evaluate_data_readiness_sensor(context)
 
     payload = _alert_payloads(caplog)[-1]
 
@@ -181,7 +185,7 @@ def test_data_readiness_sensor_rejects_invalid_provider_signal(
     message: str,
 ) -> None:
     build_sensor_context = sensor_exports["build_sensor_context"]
-    data_readiness_sensor = sensor_exports["data_readiness_sensor"]
+    evaluate_data_readiness_sensor = sensor_exports["evaluate_data_readiness_sensor"]
     context = build_sensor_context(
         resources={
             "data_readiness": _RawReadinessResource(signal),
@@ -190,7 +194,7 @@ def test_data_readiness_sensor_rejects_invalid_provider_signal(
     )
 
     with pytest.raises(TypeError, match=message):
-        data_readiness_sensor.evaluation_fn(context)
+        evaluate_data_readiness_sensor(context)
 
 
 def test_data_readiness_sensor_name(sensor_exports: dict[str, Any]) -> None:

--- a/tests/sensors/test_temporal_handoff.py
+++ b/tests/sensors/test_temporal_handoff.py
@@ -52,6 +52,7 @@ def temporal_sensor_exports() -> dict[str, Any]:
         RESOURCE_BUNDLE_RESOURCE_KEY,
         build_temporal_handoff_sensor,
         evaluate_temporal_handoff_sensor,
+        evaluate_temporal_handoff_sensor_tick,
     )
 
     return {
@@ -66,6 +67,9 @@ def temporal_sensor_exports() -> dict[str, Any]:
         "RESOURCE_BUNDLE_RESOURCE_KEY": RESOURCE_BUNDLE_RESOURCE_KEY,
         "build_temporal_handoff_sensor": build_temporal_handoff_sensor,
         "evaluate_temporal_handoff_sensor": evaluate_temporal_handoff_sensor,
+        "evaluate_temporal_handoff_sensor_tick": (
+            evaluate_temporal_handoff_sensor_tick
+        ),
     }
 
 
@@ -243,8 +247,8 @@ def test_temporal_handoff_sensor_uses_client_from_resource_bundle_and_cursors_ru
     temporal_sensor_exports: dict[str, Any],
 ) -> None:
     SkipReason = temporal_sensor_exports["SkipReason"]
-    build_temporal_handoff_sensor = temporal_sensor_exports[
-        "build_temporal_handoff_sensor"
+    evaluate_temporal_handoff_sensor_tick = temporal_sensor_exports[
+        "evaluate_temporal_handoff_sensor_tick"
     ]
     client = RecordingHandoffClient()
     context = _FakeTemporalSensorContext(
@@ -258,9 +262,13 @@ def test_temporal_handoff_sensor_uses_client_from_resource_bundle_and_cursors_ru
             _phase0_run_record("phase0-run-1", "cycle-20260416", timestamp=1.0),
         ],
     )
-    sensor_definition = build_temporal_handoff_sensor(policy=_temporal_policy())
 
-    result = sensor_definition.evaluation_fn(context)
+    result = evaluate_temporal_handoff_sensor_tick(
+        context,
+        policy=_temporal_policy(),
+        phase0_job_name="daily_cycle_phase0_job",
+        failover_job_name="daily_cycle_job",
+    )
 
     assert isinstance(result, SkipReason)
     assert "started" in result.skip_message
@@ -279,8 +287,8 @@ def test_temporal_handoff_sensor_cursor_does_not_reprocess_older_run(
     temporal_sensor_exports: dict[str, Any],
 ) -> None:
     SkipReason = temporal_sensor_exports["SkipReason"]
-    build_temporal_handoff_sensor = temporal_sensor_exports[
-        "build_temporal_handoff_sensor"
+    evaluate_temporal_handoff_sensor_tick = temporal_sensor_exports[
+        "evaluate_temporal_handoff_sensor_tick"
     ]
     client = RecordingHandoffClient()
     cursor = json.dumps(
@@ -303,9 +311,13 @@ def test_temporal_handoff_sensor_cursor_does_not_reprocess_older_run(
             _phase0_run_record("phase0-run-1", "cycle-20260416", timestamp=1.0),
         ],
     )
-    sensor_definition = build_temporal_handoff_sensor(policy=_temporal_policy())
 
-    result = sensor_definition.evaluation_fn(context)
+    result = evaluate_temporal_handoff_sensor_tick(
+        context,
+        policy=_temporal_policy(),
+        phase0_job_name="daily_cycle_phase0_job",
+        failover_job_name="daily_cycle_job",
+    )
 
     assert isinstance(result, SkipReason)
     assert "no successful daily_cycle_phase0_job run is ready" in result.skip_message
@@ -317,8 +329,8 @@ def test_temporal_handoff_sensor_processes_unseen_run_at_same_timestamp(
     temporal_sensor_exports: dict[str, Any],
 ) -> None:
     SkipReason = temporal_sensor_exports["SkipReason"]
-    build_temporal_handoff_sensor = temporal_sensor_exports[
-        "build_temporal_handoff_sensor"
+    evaluate_temporal_handoff_sensor_tick = temporal_sensor_exports[
+        "evaluate_temporal_handoff_sensor_tick"
     ]
     client = RecordingHandoffClient()
     context = _FakeTemporalSensorContext(
@@ -332,11 +344,25 @@ def test_temporal_handoff_sensor_processes_unseen_run_at_same_timestamp(
             _phase0_run_record("phase0-run-1", "cycle-20260416", timestamp=2.0),
         ],
     )
-    sensor_definition = build_temporal_handoff_sensor(policy=_temporal_policy())
 
-    first_result = sensor_definition.evaluation_fn(context)
-    second_result = sensor_definition.evaluation_fn(context)
-    third_result = sensor_definition.evaluation_fn(context)
+    first_result = evaluate_temporal_handoff_sensor_tick(
+        context,
+        policy=_temporal_policy(),
+        phase0_job_name="daily_cycle_phase0_job",
+        failover_job_name="daily_cycle_job",
+    )
+    second_result = evaluate_temporal_handoff_sensor_tick(
+        context,
+        policy=_temporal_policy(),
+        phase0_job_name="daily_cycle_phase0_job",
+        failover_job_name="daily_cycle_job",
+    )
+    third_result = evaluate_temporal_handoff_sensor_tick(
+        context,
+        policy=_temporal_policy(),
+        phase0_job_name="daily_cycle_phase0_job",
+        failover_job_name="daily_cycle_job",
+    )
 
     assert isinstance(first_result, SkipReason)
     assert isinstance(second_result, SkipReason)
@@ -358,8 +384,8 @@ def test_temporal_handoff_sensor_legacy_cursor_stops_at_processed_run(
     temporal_sensor_exports: dict[str, Any],
 ) -> None:
     SkipReason = temporal_sensor_exports["SkipReason"]
-    build_temporal_handoff_sensor = temporal_sensor_exports[
-        "build_temporal_handoff_sensor"
+    evaluate_temporal_handoff_sensor_tick = temporal_sensor_exports[
+        "evaluate_temporal_handoff_sensor_tick"
     ]
     client = RecordingHandoffClient()
     context = _FakeTemporalSensorContext(
@@ -374,9 +400,13 @@ def test_temporal_handoff_sensor_legacy_cursor_stops_at_processed_run(
             _phase0_run_record("phase0-run-1", "cycle-20260416", timestamp=1.0),
         ],
     )
-    sensor_definition = build_temporal_handoff_sensor(policy=_temporal_policy())
 
-    result = sensor_definition.evaluation_fn(context)
+    result = evaluate_temporal_handoff_sensor_tick(
+        context,
+        policy=_temporal_policy(),
+        phase0_job_name="daily_cycle_phase0_job",
+        failover_job_name="daily_cycle_job",
+    )
 
     assert isinstance(result, SkipReason)
     assert "no successful daily_cycle_phase0_job run is ready" in result.skip_message

